lib/possible_bipartition.rb

```diff
@@ -1,4 +1,45 @@
 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+  return true if dislikes.empty?
+  graph_queue = []
+  searched = []
+  
+  color_array = []
+  dislikes.length.times do
+    color_array << 0
+  end
+  
+  dislikes.each_with_index { |enemies, dog|
+    graph_queue << dog
+  }
```

Comment:
Instead of adding each node to the Queue in the beginning, maybe make that while loop below do something like this:

```ruby
dislikes.each_with_index do |enemies, dog|
  unless searched[dog] # make searched a hash
    graph_queue << dog
     current_color = # whatever starting color you're on
    while !graph_queue.empty?
     # Then your code continues with your while loop
```